src/modules/aws_vpc/mappers/dhcp_options.ts

```diff
@@ -0,0 +1,165 @@
+import { raw } from 'express';
+import isEqual from 'lodash.isequal';
+
+import {
+  CreateDhcpOptionsCommandInput,
+  DhcpOptions as AWSDhcpOptions,
+  EC2,
+  NewDhcpConfiguration,
+  paginateDescribeDhcpOptions,
+  Tag,
+} from '@aws-sdk/client-ec2';
+
+import { AwsVpcModule } from '..';
+import { AWS, crudBuilder, crudBuilderFormat, paginateBuilder } from '../../../services/aws_macros';
+import { Context, Crud, MapperBase } from '../../interfaces';
+import { DhcpOptions } from '../entity';
+import { eqTags } from './tags';
+
+export class DhcpOptionsMapper extends MapperBase<DhcpOptions> {
+  module: AwsVpcModule;
+  entity = DhcpOptions;
+  equals = (a: DhcpOptions, b: DhcpOptions) =>
+    Object.is(a.dhcpConfigurations?.length, b.dhcpConfigurations?.length) &&
+    isEqual(a.dhcpConfigurations, b.dhcpConfigurations) &&
+    eqTags(a.tags, b.tags);
+
+  async dhcpOptionsMapper(e: AWSDhcpOptions, region: string, ctx: Context) {
+    if (!e.DhcpOptionsId) return undefined;
+    const out = new DhcpOptions();
+    out.dhcpOptionsId = e.DhcpOptionsId;
+    out.dhcpConfigurations = e.DhcpConfigurations as any;
+
+    if (e.Tags?.length) {
+      const tags: { [key: string]: string } = {};
+      e.Tags.filter((t: any) => !!t.Key && !!t.Value).forEach((t: any) => {
+        tags[t.Key as string] = t.Value as string;
+      });
```

Comment:
Nit: I think this would also work and not require `as` casting.

```suggestion
      e.Tags.filter((t: { Key: string, Value: string }) => !!t.Key && !!t.Value).forEach((t: { Key: string, Value: string }) => {
        tags[t.Key] = t.Value;
      });
```

The second explicit type on the `.forEach` may or may not be necessary depending on how clever `tsc` is.